src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/ProvidePublicParameterlessSafeHandleConstructor.Fixer.cs

```diff
@@ -0,0 +1,107 @@
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class ProvidePublicParameterlessSafeHandleConstructorFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(ProvidePublicParameterlessSafeHandleConstructorAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(context.Document);
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode enclosingNode = root.FindNode(context.Span);
+            SyntaxNode declaration = generator.GetDeclaration(enclosingNode);
+            if (declaration == null)
+            {
+                return;
+            }
+
+            SemanticModel model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            ISymbol symbol = model.GetDeclaredSymbol(declaration, context.CancellationToken);
+            if (symbol is IMethodSymbol method)
+            {
+                Diagnostic diagnostic = context.Diagnostics.First();
+                string title = MicrosoftNetCoreAnalyzersResources.ProvidePublicParameterlessSafeHandleConstructorTitle;
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        title,
+                        async ct => await MakeParameterlessConstructorPublic(declaration, context.Document, context.CancellationToken).ConfigureAwait(false),
+                        equivalenceKey: title),
+                    diagnostic);
+            }
+            else if (symbol is INamedTypeSymbol type)
+            {
+                bool baseTypeHasAccessibleParameterlessConstructor = false;
+                foreach (var constructor in type.BaseType.InstanceConstructors)
+                {
+                    if (constructor.Parameters.Length == 0 &&
+                        model.Compilation.IsSymbolAccessibleWithin(constructor, type, type))
+                    {
+                        baseTypeHasAccessibleParameterlessConstructor = true;
+                        break;
+                    }
+                }
+
+                if (baseTypeHasAccessibleParameterlessConstructor)
```

Comment:
Is there any test where this condition will be false?
If it's false, then no codefix will be provided, I don't see any test with a diagnostic but no codefix.